include/xtensor/xtiny.hpp

```diff
@@ -0,0 +1,1999 @@
+/***************************************************************************
+* Copyright (c) 2017, Ullrich Koethe                                       *
+*                                                                          *
+* Distributed under the terms of the BSD 3-Clause License.                 *
+*                                                                          *
+* The full license is in the file LICENSE, distributed with this software. *
+****************************************************************************/
+
+#ifndef XTENSOR_XTINY_HPP
+#define XTENSOR_XTINY_HPP
+
+#include <array>
+#include <type_traits>
+#include <iosfwd>
+#include <algorithm>
+#include <memory>
+#include <iterator>
+#include <utility>
+#include <tuple>  // std::ignore
+
+#include "xconcepts.hpp"
+#include "xexception.hpp"
+#include "xutils.hpp"
+#include "xbuffer_adaptor.hpp"
+#include "xstorage.hpp"
+
+namespace xt
+{
+    /*****************/
+    /* prerequisites */
+    /*****************/
+
+    using index_t = std::ptrdiff_t;
+
+    static const index_t runtime_size  = -1;
```

Comment:
I think replacing `static const` with `constexpr static` would be nice.

Reply:
changed.